tests/iostability/test_iostability_happy_path.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2022 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+"""Test Suite for IO stability Happy Path workloads."""
+import logging
+import os
+import time
+from datetime import datetime, timedelta
+
+import pytest
+
+from commons import configmanager
+from commons.constants import K8S_SCRIPTS_PATH
+from commons.helpers.pods_helper import LogicalNode
+from commons.params import LATEST_LOG_FOLDER
+from commons.utils import support_bundle_utils, assert_utils, system_utils
+from config import CMN_CFG
+from config.s3 import S3_CFG
+from conftest import LOG_DIR
+from libs.s3 import ACCESS_KEY, SECRET_KEY
+from scripts.s3_bench import s3bench
+
+
+class TestIOWorkload:
+    """Test suite for IO workloads."""
+
+    @classmethod
+    def setup_class(cls):
+        """Setup class."""
+
+        cls.log = logging.getLogger(__name__)
+        cls.num_nodes = len(CMN_CFG["nodes"])
+        cls.master_node_list = []
+        cls.worker_node_list = []
+        for node in CMN_CFG['nodes']:
+            node_obj = LogicalNode(hostname=node["hostname"],
+                                   username=node["username"],
+                                   password=node["password"])
+            if node["node_type"].lower() == "master":
+                cls.master_node_list.append(node_obj)
+            else:
+                cls.worker_node_list.append(node_obj)
+        cls.test_cfg = configmanager.get_config_wrapper(fpath="config/iostability_test.yaml")
+        cls.setup_type = CMN_CFG["setup_type"]
+        cls.test_completed = False
+
+    def teardown_class(self):
+        """Teardown class"""
+        if not self.test_completed:
+            self.log.info("Test Failure observed, collecting support bundle")
+            path = os.path.join(LOG_DIR, LATEST_LOG_FOLDER)
+            resp = support_bundle_utils.collect_support_bundle_k8s(local_dir_path=path,
+                                                                   scripts_path=K8S_SCRIPTS_PATH)
+            assert_utils.assert_true(resp)
+
+    def execute_workload_distribution(self, distribution, clients, total_obj,
```

Comment:
can you reuse the execute_workload_distribution function in degraded path? Looks like similar function being copied to another file. Best is to extract it in a lib file.